### PR TITLE
Allow filtering on plugin metadata to avoid inappropriate plugins

### PR DESCRIPTION
**Pull request by SimonWalton-HiFi**
Ticket: https://highfidelity.manuscript.com/f/cases/10181/Plugins-for-Interface-are-being-loaded-by-ACs

Add a simple functor-based filtering mechanism to our PluginManager. The audio mixer uses this to only select audio codecs by limiting to DLLs that have the string 'codec' in their Qt-provided metadata name. This filtering is doing before loading to avoid the crash discussed in the ticket.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16705/HighFidelity-Beta-PR14763-74748b1.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16705/HighFidelity-Beta-PR14763-74748b1.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16705/HighFidelity-Beta-Interface-PR14763-74748b1.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16705/HighFidelity-Beta-PR14763-74748b1.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16705/HighFidelity-Beta-Interface-PR14763-74748b1.exe

File: libraries/plugins/src/plugins/PluginManager.cpp [hunk 1]
```diff
@@ -106,15 +108,25 @@ const LoaderList& getLoadedPlugins() {
...
+#endif
+                if (isDisabled(pluginMetaData)) {
                     qCWarning(plugins) << "Plugin" << qPrintable(plugin) << "is disabled";
                     // Skip this one, it's disabled
                     continue;
                 }
-                if (getPluginInterfaceVersionFromMetaData(loader->metaData()) != HIFI_PLUGIN_INTERFACE_VERSION) {
+
+                if (!_pluginFilter(pluginMetaData)) {
+                    qCWarning(plugins) << "Plugin" << qPrintable(plugin) << "doesn't pass provided filter";
```
[the remark this answers is not in the document]
Comment (SimonWalton-HiFi) [reply libraries/plugins/src/plugins/PluginManager.cpp]: Fixed.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16712/HighFidelity-Beta-PR14763-cc6caa6.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16712/HighFidelity-Beta-PR14763-cc6caa6.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16712/HighFidelity-Beta-Interface-PR14763-cc6caa6.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16712/HighFidelity-Beta-PR14763-cc6caa6.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16712/HighFidelity-Beta-Interface-PR14763-cc6caa6.exe

**Comment by qlipdahlin** [comment]
Currently Testing

**Comment by qlipdahlin** [comment]
PR Passes.

Test on Master and found in Audio-Mixer log the following line:
[01/24 15:23:57] [DEBUG] [hifi.plugins] [8772] [assignment-client] Attempting plugin hifiKinect.dll
[01/24 15:23:57] [DEBUG] [hifi.plugins] [8772] [assignment-client] Plugin hifiKinect.dll failed to load:

Repeated test on PR build:
[01/24 15:53:38] [DEBUG] [hifi.plugins] [10652] [assignment-client] Attempting plugin hifiKinect.dll
[01/24 15:53:38] [DEBUG] [hifi.plugins] [10652] [assignment-client] Plugin hifiKinect.dll doesn't pass provided filter

Shows plugin that failed to load previously now fails because it doesn't pass provided filter.  PR Passes.

**Pull request merged**